internal/services/datadog/datadog_monitor_tag_rule_resource.go

```diff
@@ -363,3 +369,24 @@ func flattenFilteringTags(input *[]rules.FilteringTag) []interface{} {
 	}
 	return results
 }
+
+func isDefaultSettings(input *rules.MonitoringTagRules) bool {
+	if input == nil {
+		return false
+	}
+
+	if name := input.Name; name != nil && *input.Name == "default" {
+		if props := input.Properties; props != nil {
+			logRules := props.LogRules
+			metricRules := props.MetricRules
+
+			if logRules != nil && metricRules != nil {
+				if (logRules.SendAadLogs != nil && !*logRules.SendAadLogs) && (logRules.SendSubscriptionLogs != nil && !*logRules.SendSubscriptionLogs) && (logRules.SendResourceLogs != nil && !*logRules.SendResourceLogs) && (logRules.FilteringTags != nil && len(*logRules.FilteringTags) == 0) && (metricRules.FilteringTags != nil && len(*metricRules.FilteringTags) == 0) {
+					return true
+				}
+			}
+		}
+	}
```

Comment:
we can simplify this by shifting left, for example:

```suggestion
	if input.Name == nil || strings.EqualsFold(*input.Name, "default") {
		return false
	}
	if input.Properties == nil || input.Properties.LogRules == nil || input.Properties.MetricRules == nil {
		return false
	}
	logRules := input.Properties.LogRules 
	metricRules := input.Properties.MetricRules
	result := logRules.SendAadLogs != nil && !*logRules.SendAadLogs) &&
	(logRules.SendSubscriptionLogs != nil && !*logRules.SendSubscriptionLogs) &&
	 (logRules.SendResourceLogs != nil && !*logRules.SendResourceLogs) &&
	 (logRules.FilteringTags != nil && len(*logRules.FilteringTags) == 0) &&
	 (metricRules.FilteringTags != nil && len(*metricRules.FilteringTags) == 0)
	return result
```

Reply:
Updated